### PR TITLE
xplat perf improvements

### PR DESCRIPTION
**Pull request by obastemur**
#### osx: cache expensive threadId retrieval
`pthread_threadid_np(pthread_self(), &tid);` call is very expensive. Cache expensive threadId retrieval instead. Same commit also removes unused `THREADGetThreadProcessId`

#### xplat: Improve PAL wmemcmp perf.
Since we know the char count, comparing with larger blocks reduces the total cycles to spend.

### CODE REVIEW

File: pal/src/cruntime/wchar.cpp [hunk 1]
```diff
@@ -1241,14 +1241,21 @@ PAL_wmemcmp(
 
     if (string1 == string2) return diff;
 
-    for (i = 0; i < count; i++)
+    constexpr size_t blockSize = sizeof(size_t) / sizeof(char16_t);
+    const     size_t *num1     = (const size_t*)(string1);
+    const     size_t *num2     = (const size_t*)(string2);
+
+    while( (count >= blockSize * wi) && !(num1[wi] - num2[wi]) ) ++wi;
```
Comment (jianchun) [inline pal/src/cruntime/wchar.cpp]: >count >= blockSize * wi [](start = 12, length = 23)

`count >= blockSize * wi` can result in out-of-bound read at the end (when they equal).

Comment (obastemur) [inline pal/src/cruntime/wchar.cpp]: Yep, a silly edit issue on my end. That's why the test fails on CI. Thanks for pointing it out though.

**Comment by obastemur** [comment]
retargeted to release/1.6

File: pal/src/cruntime/wchar.cpp [hunk 2]
```diff
@@ -1241,14 +1241,21 @@ PAL_wmemcmp(
 
     if (string1 == string2) return diff;
 
-    for (i = 0; i < count; i++)
+    constexpr size_t blockSize = sizeof(size_t) / sizeof(char16_t);
+    const     size_t *num1     = (const size_t*)(string1);
+    const     size_t *num2     = (const size_t*)(string2);
+
+    while( (count > blockSize * wi) && !(num1[wi] - num2[wi]) ) ++wi;
```
Comment (jianchun) [inline pal/src/cruntime/wchar.cpp]: `!(num1[wi] - num2[wi])` ==> `num1[wi] == num2[wi]` would be easier to read?

**Comment by obastemur** [comment]
@jianchun Thanks for the review

**Pull request merged**